scripts/new_project.kts

```diff
@@ -1,25 +1,68 @@
+#!/usr/bin/env kscript
+
+@file:MavenRepository("central", "https://repo.maven.apache.org/maven2/")
+@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.1")
+
+import kotlinx.coroutines.*
+import java.io.File
+import kotlin.coroutines.coroutineContext
+
 object NewProject {
 
     private const val KEY_APP_NAME = "app-name"
     private const val KEY_PACKAGE_NAME = "package-name"
     private const val ARGUMENT_DELIMITER = "="
+    private const val COROUTINE_FOLDER_NAME = "CoroutineTemplate"
+    private const val COROUTINE_PACKAGE_THIRD_SEG = "coroutine"
 
     private var appName = ""
     private var packageName = ""
+    private var appNameWithoutSpace = ""
 
-    fun generate(args: Array<String>) {
+    fun generate(args: Array<String>) = runBlocking {
         handleArguments(args)
+
+        showMessage("=> 🐢 Initializing new project...")
+        copyFiles(to = appNameWithoutSpace)
+        /*
+         * Set gradlew file as executable
+         * Copying file from one folder to another lost it's previous execution right.
+         * Need to set gradlew as executable to run
+         */
+        File("$appNameWithoutSpace/gradlew")?.setExecutable(true)
     }
 
     private fun handleArguments(args: Array<String>) {
         args.forEach {
             val (key, value) = it.split(ARGUMENT_DELIMITER)
             when (key) {
-                KEY_APP_NAME -> appName = value
+                KEY_APP_NAME -> {
+                    appName = value
+                    appNameWithoutSpace = appName.trim().replace(" ", "")
+                }
                 KEY_PACKAGE_NAME -> packageName = value
             }
         }
     }
+
+    private fun showMessage(message: String?) {
+        runBlocking {
+            println("\n${message?.trimIndent()}\n")
+        }
+    }
+
+    private suspend fun copyFiles(to: String) {
+        withContext(coroutineContext) {
+            async(Dispatchers.IO) {
```

Comment:
By using coroutines, does it mean the script can execute faster than without? 🙏 

If yes, should we move this to `fun generate(...)` instead? So all functions are executed on `Dispatchers.IO`

Reply:
Hmmmm, please allow me to search a bit more on this. I'm not so familiar with using coroutines in a script context. Do you have any reference/link I can look into by any chance? 🙏

Reply:
@kaungkhantsoe when we run long-running tasks in an interface, I agree that we should use `IO` thread. But in this case, `copyRecursively` function is using `for loop` for the execution, and other complex tasks are neither executed nor are running on an interface. I think we can get rid of `coroutines` in the script 🙇  Please let me know your opinion too @nimblehq/android-chapter

Reply:
@kaungkhantsoe Could you please share a screen recording that demonstrates why you chose to go with coroutines? I recall you mentioned to me that you faced an issue if we don't use coroutines 🙏